fiaas_deploy_daemon/deployer/kubernetes/role_binding.py

```diff
@@ -0,0 +1,100 @@
+import logging
+
+from k8s.models.common import ObjectMeta
+from k8s.models.role_binding import RoleBinding, RoleRef, Subject
+from k8s.client import NotFound
+from fiaas_deploy_daemon.specs.models import AppSpec
+from fiaas_deploy_daemon.deployer.kubernetes.owner_references import OwnerReferences
+from fiaas_deploy_daemon.tools import merge_dicts
+
+LOG = logging.getLogger(__name__)
+
+
+class RoleBindingDeployer:
+    def __init__(self, config, owner_references):
+        self._owner_references: OwnerReferences = owner_references
+        self._list_of_roles = config.list_of_roles
+        self._list_of_cluster_roles = config.list_of_cluster_roles
+
+    def deploy(self, app_spec: AppSpec, labels):
+        custom_annotations = {}
+        custom_labels = labels
+        custom_labels = merge_dicts(app_spec.labels.role_binding, custom_labels)
+        custom_annotations = merge_dicts(app_spec.annotations.role_binding, custom_annotations)
+        role_bindings = RoleBinding.find(name=app_spec.name, namespace=app_spec.namespace)
+        self._clean_not_needed_role_bindings(role_bindings)
+        counter = 1
+        self._create_role_bindings(app_spec, self._list_of_roles, "Role", counter, custom_annotations, custom_labels, role_bindings)
+        self._create_role_bindings(app_spec, self._list_of_cluster_roles, "ClusterRole", counter, custom_annotations, custom_labels,
+                                   role_bindings)
+
+    def _create_role_bindings(self, app_spec: AppSpec, roles_list, role_kind, counter, custom_annotations, custom_labels,
+                              role_bindings):
+        LOG.info("Creating RoleBindings for %s", app_spec.name)
+        namespace = app_spec.namespace
+
+        service_account_name = app_spec.name
+
+        for role_name in roles_list:
+            role_binding = self._find_role_in_role_bindings(role_kind, role_name, role_bindings)
+            if role_binding:
+                if self._owned_by_fiaas(role_binding):
+                    role_binding.metadata = ObjectMeta(
+                        name=role_binding.metadata.name, namespace=namespace, labels=custom_labels, annotations=custom_annotations
+                    )
+                    role_ref = RoleRef(kind=role_kind, apiGroup="rbac.authorization.k8s.io", name=role_name)
+                    subject = Subject(kind="ServiceAccount", name=service_account_name, namespace=namespace)
+                    role_binding.roleRef = role_ref
+                    role_binding.subjects = [subject]
+                    self._owner_references.apply(role_binding, app_spec)
+                    role_binding.save()
```

Comment:
There's nine lines here that are identical and need to be kept in sync with below. I think that's enough to warrant extracting it to a separate method.

Reply:
Extracted to a separate method and moved it